### PR TITLE
chore(launch-6.0-markets): support 6.0 market launch flow

### PR DESCRIPTION
**Pull request by jaredvu**
| before | after |
| --- | --- |
| Screen Shot 2024-08-27 at 4 14 12 PM | Screen Shot 2024-08-27 at 8 42 05 PM |
| Screen Shot 2024-08-27 at 8 42 44 PM | Screen Shot 2024-08-27 at 8 43 16 PM |

- Add `BUFFI-USD` metadata
- add new liquidity tiers
- add asset helper functions w/ tests

---

## Views

* `<NewMarketSelectionStep>`, `<NewMarketPreviewStep>`
  * Use new asset utils to display simple asset/market tag

* `<NewMarketMessageDetailDialog>`
  * Add rules to `<Details>` to handle text overflow hiding label

## Components

* `<Details>`
  * Add grid-template-column rules so that the label isn't hidden

* `<AssetIcon>`
  * add `buffi.png`

## Functions

* new `lib/assetUtils` + tests
  * asset helper functions

## Hooks

* `hooks/usePotentialMarkets`
  * Add new liquidity tier data

## Packages

* `vitest`
  * updated: v0.32.2 -> v1.4.0

---

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -20386,6 +20361,13 @@ packages:
       yocto-queue: 1.0.0
     dev: true
 
+  /p-limit@5.0.0:
```
Comment (tyleroooo) [inline pnpm-lock.yaml:20364]: now we have like 4 copies of this lib for some reason

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -23608,6 +23585,12 @@ packages:
   /type-detect@4.0.8:
     resolution: {integrity: sha512-0fr/mIH1dlO+x7TlcMy+bIDqKPsw/70tVyeHW787goQjhmqaZe10uwLujubK9q9Lg6Fiho1KUKDYz0Z7k7g5/g==}
     engines: {node: '>=4'}
+    dev: false
+
+  /type-detect@4.1.0:
```
Comment (tyleroooo) [inline pnpm-lock.yaml:23590]: and this one

File: public/currencies/buffi.png
Comment (tyleroooo) [inline public/currencies/buffi.png:1]: thanks i hate it

File: src/views/dialogs/NewMarketMessageDetailsDialog.tsx [hunk 3]
```diff
@@ -341,7 +342,19 @@ const $Tabs = styled(ToggleGroup)`
   overflow-x: auto;
 ` as typeof ToggleGroup;
 
-const $Details = tw(Details)`[--details-item-height:1.5rem]`;
+const $Details = styled(Details)`
+  ${tw`[--details-item-height:1.5rem]`}
```
Comment (tyleroooo) [inline src/views/dialogs/NewMarketMessageDetailsDialog.tsx:346]: lol at this point we might as well expand it to normal css but I'm okay either way. its kinda cute

File: src/lib/assetUtils.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/**
+ *
+ * @param fullBaseAsset contains base asset, occassionally accompanied (comma-separated) by dex and address. i.e. 'baseAsset,dex,address'
+ * @param part
+ * @returns base asset or dex or address from the full base asset.
+ */
+export const getDisplayableAssetFromBaseAsset = (
```
Comment (tyleroooo) [inline src/lib/assetUtils.ts:7]: these functions might be a kinda good candidate for memoization - just throw a 1000 entry lru cache in there and save some string processing time

**Review by tyleroooo — COMMENTED** [review]
looks good, just a few nits

**Review by tyleroooo — APPROVED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (jaredvu) [reply pnpm-lock.yaml:20364]: 4 different version :/

Comment (jaredvu) [reply pnpm-lock.yaml:20364]: do you think I should delete pnpm-lock and regen?

Comment (tyleroooo) [reply pnpm-lock.yaml:20364]: nawww but maybe we should do another cleanup + remove duplicates run.

**Review by tyleroooo — APPROVED** [review]

**Pull request merged**